lib/sdam/server.js

```diff
@@ -70,15 +85,27 @@ class Server extends EventEmitter {
     }
 
     // create a pool
-    this.s.pool = new Pool(this, Object.assign(this.s.options, options, { bson: this.s.bson }));
+    const addressParts = this.description.address.split(':');
+    const poolOptions = Object.assign(
+      { host: addressParts[0], port: parseInt(addressParts[1], 10) },
+      this.s.options,
+      options,
+      { bson: this.s.bson }
+    );
+
+    // NOTE: this should only be the case if we are connecting to a single server
+    poolOptions.reconnect = true;
```

Comment:
short answer: yes. longer answer: in order to provide parity with the existing topologies (a primary goal until we deprecate them, and make the "Big Switch"), we need to keeps reconnect enabled. Eventually this will be completely removed in favor of retryable reads and writes